helm/templates/deployment.yaml

```diff
@@ -20,7 +20,7 @@ spec:
         env:                
         - name: NODE_ENV
           value: {{ .Values.environment }}
-        - name:MESSAGE_QUEUE
+        - name: MESSAGE_QUEUE
```

Comment:
There is no `MESSAGE_QUEUE` environment variable used in the application.
The values files needs to and allow the following env vars to be set so the service can be configured during deployment:
```
MESSAGE_QUEUE_HOST
MESSAGE_QUEUE_PORT
MESSAGE_QUEUE_TRANSPORT
CALCULATION_QUEUE_ADDRESS
CALCULATION_QUEUE_USER
CALCULATION_QUEUE_PASSWORD
PAYMENT_QUEUE_ADDRESS
PAYMENT_QUEUE_USER
PAYMENT_QUEUE_PASSWORD
```

Ideally the helm templates should be updated to adhere to the standards.

Reply:
the port in the values files and the service.yaml are not required as the application does not expose a web service

Reply:
Thanks, missed this config.  Do we have a message queue we can configure these values to in the PR?

Reply:
For now we can use the service bus payment and calculation queues, alternatively a PR test ones could be set up, there's some notes here as to how the 'real' ones were created https://eaflood.atlassian.net/secure/RapidBoard.jspa?rapidView=533&modal=detail&selectedIssue=FPD-337&search=bus

Reply:
I think it makes sense to longer term to setup a PR specific queue so PRs don't interfere with the actual service.

Reply:
I've configured the helm chart to match the standard and it has deployed to AKS correctly.

I have been thinking about the message queue though, currently it's just using artemis within the cluster which isn't ideal.  I was about to change it to the service bus but I'm not sure where is best to store the credentials securely without configuring Key Vault.  Thoughts?

Reply:
For the user service I've stored database server and credentials as pipeline variables

Reply:
Aren't the pipeline variables visible in the yaml and publically through GitHub though?

Reply:
Or have you set them in the server variables in ADO?  (I presume you can still have both when using yaml)

Reply:
Got it, saved them as private variables in ADO.  They are now overridden in the PR deployment.